gcc/rust/lex/rust-lex.h

```diff
@@ -223,22 +196,56 @@ class Lexer
   static const int max_column_hint = 80;
 
   // Input source wrapper thing.
-  struct InputSource
+  class InputSource
   {
+  public:
+    virtual ~InputSource () {}
+
+    // Overload operator () to return next char from input stream.
+    virtual int operator() () = 0;
```

Comment:
Oof :DD Very cplusplussy! If everyone is fine with it, I won't opppose it, but I personally think this would be a little more clear :)
```suggestion
    virtual int next () = 0;
```

I know I'm going to be confused when I'm gonna see `raw_input_source()` return a character

Reply:
yeah i agree let me see how painful this is to update otherwise i might leave it as a good first-pr to refactor this

Reply:
Looks like it wasn't too bad to update this.